vdr/src/contracts/cl/types/credential_definition.rs

```diff
@@ -24,6 +24,45 @@ pub enum CredentialDefinitionTypes {
     CL,
 }
 
+impl CredentialDefinition {
+    pub fn id(&self) -> CredentialDefinitionId {
+        CredentialDefinitionId::build(&self.issuer_id, &self.schema_id.to_string(), &self.tag)
+    }
+
+    pub fn matches_id(&self, expected_id: &CredentialDefinitionId) -> VdrResult<()> {
+        let actual_id = self.id();
+
+        if expected_id.to_string() != actual_id.to_string() {
+            return Err(VdrError::InvalidCredDef(format!(
+                "Id built from cred_def: {} != provided id: {}",
```

Comment:
Credential Definition ID `{}` does not match to requested `{}`

Reply:
Changed description